### PR TITLE
fix: Fix lot of nullable annotation issues 

### PR DESCRIPTION
**Pull request by dr1rrb**
## Bugfix
fix: Fix lot of nullable annotation issues (appears when updating to newer SDK version)

## What is the current behavior?
Compliant with netstd 2.0

## What is the new behavior?
Compliant with net 8.0

## PR Checklist
- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md). (for bug fixes / features) 
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/Uno.Extensions.Reactive/Collections/Facades/Adapters/ListToUntypedList.cs [hunk 1]
```diff
@@ -47,16 +47,16 @@ public ListToUntypedList(IList<T> inner)
 	}
 
 	/// <inheritdoc />
-	public bool Contains(object value)
-		=> _inner.Contains((T)value);
+	public bool Contains(object? value)
+		=> value is T t && _inner.Contains(t);
```
Comment (carldebilly) [inline src/Uno.Extensions.Reactive/Collections/Facades/Adapters/ListToUntypedList.cs:51]: Cannot contain `null`? (potential behavior change)

File: src/Uno.Extensions.Reactive/Collections/Facades/Map/ConverterEqualityComparer.cs [hunk 2]
```diff
@@ -22,10 +22,16 @@ public ConverterEqualityComparer(IConverter<TFrom, TTo> converter, IEqualityComp
...
-	public bool Equals(TTo left, TTo right) 
-		=> _comparer.Equals(_converter.ConvertBack(left), _converter.ConvertBack(right));
+	public bool Equals(TTo? left, TTo? right)
+		=> (left, right) switch
+		{
+			(null, null) => true,
+			(null, _) => false,
+			(_, null) => false,
+			_ => _comparer.Equals(_converter.ConvertBack(left), _converter.ConvertBack(right))
+		};
```
Comment (carldebilly) [inline src/Uno.Extensions.Reactive/Collections/Facades/Map/ConverterEqualityComparer.cs:32]: Potential perf hit.

**Review by carldebilly — APPROVED** [review]

File: src/Uno.Extensions.Reactive/Collections/Facades/Map/ConverterEqualityComparer.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.Extensions.Reactive/Collections/Facades/Map/ConverterEqualityComparer.cs:32]: Those are ref equals comparison, it should not have any perf hit.

> When you match an expression against null, the compiler guarantees that no user-overloaded == or != operator is invoked.

File: src/Uno.Extensions.Reactive/Collections/Facades/Adapters/ListToUntypedList.cs (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.Extensions.Reactive/Collections/Facades/Adapters/ListToUntypedList.cs]: Good catch, fixed 👍

**Pull request merged**